### ISSUE TITLE
During SSR, should render into the document, not the document.body

### ISSUE THREAD

**Issue by matthewp** [donejs/autorender#55]
Currently done-autorender renders into the `document.body` during server-side rendering, which leads to weird HTML like:

```html
<html>
<head></head>
<body>
  <html>
    <head><title>App</title></head>
    <body>... stuff </body>
  </html>
</body>
</html>
```

Gross.  I don't remember the exact reason why I did it this way. I think the reason was probably because most `index.stache` files contain an `<html>` tag, and you can't appendChild directly into the document.

I think the solution is to instead replace the `document.documentElement` with the rendered `<html>`, if that's possible.

**Comment by matthewp** [comment]
You're probably thinking of client-side reattachment, which does do that.  SSR will be the same soon though.

**Comment by matthewp** [comment]
The reason why the code is not exactly the same here is because ssr and reattachment are doing two different things.

SSR is calling the renderer, and immediately attaching the fragment into a (new) document. done-ssr then does the waiting using its own Zone logic.

Reattachment calls the renderer, creates a Zone and waits for the fragment to be fully rendered, and *then* inserts it into the (real) document by pulling out the head and body.

**Comment by matthewp** [comment]
Closed by #57
